numba/core/analysis.py

```diff
@@ -347,6 +347,21 @@ def prune_by_value(branch, condition, blk, *conds):
         taken = do_prune(take_truebr, blk)
         return True, taken
 
+    def prune_by_predicate(branch, pred, blk):
+        try:
+            # Just to prevent accidents, whilst already guarded, ensure this
+            # is an ir.Const
+            if not isinstance(pred, (ir.Const, ir.FreeVar, ir.Global)):
+                raise TypeError
+            take_truebr = bool(pred.value)
+        except Exception:
```

Comment:
Would it only raise `TypeError`? Better to be more specific.

Reply:
I'm not sure it matters? I picked `TypeError` as it's vaguely suitable, the idea here is that if it's not an accepted type then to just abort doing a prune, the mechanism to achieve this state is to throw an exception.